File: webflux-dialogue/src/test/java/com/study/webflux/rag/config/annotation/ReactiveRepositoryTest.java
package com.study.webflux.rag.config.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
import org.springframework.test.context.ActiveProfiles;

import org.junit.jupiter.api.Tag;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@DataMongoTest
@ActiveProfiles("test")
@Tag("repository")
public @interface ReactiveRepositoryTest {

}

File: infrastructure/src/test/kotlin/com/miyou/app/config/annotation/ReactiveRepositoryTest.kt
package com.miyou.app.config.annotation

import org.junit.jupiter.api.Tag
import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest
import org.springframework.test.context.ActiveProfiles

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@DataMongoTest
@ActiveProfiles("test")
@Tag("repository")
annotation class ReactiveRepositoryTest
